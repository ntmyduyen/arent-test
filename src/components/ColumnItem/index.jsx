import { WrapColumnItem } from "./index.styles"

const ColumnItem = ({ url, dateTime, title, tags }) => {
    return (
        <WrapColumnItem url={url}>
            <div className="wrap-img">
                <div className="date-time">{dateTime}</div>
            </div>
            <div className="title" title={title}>{title}</div>
            <div className="tags">
                {
                    tags.map((tag, index) => (
                        <div className="tag" key={index}>#{tag}</div>
                    ))
                }
            </div>
        </WrapColumnItem>
    )
}
export default ColumnItem