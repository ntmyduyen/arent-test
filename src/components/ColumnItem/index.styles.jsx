import styled from 'styled-components'

const WrapColumnItem = styled.div`
    width: 234px;
    .wrap-img {
        width: inherit;
        height: 144px;
        background: url(${props => props.url});
        background-size: cover;
        background-position: center;
        background-repeat: no-repeat;
        position: relative;

        .date-time {
            position: absolute;
            bottom: 0;
            left: 0;
            font-family: var(--secondary-font);
            font-style: normal;
            font-weight: 400;
            font-size: 15px;
            line-height: 30px;
            color: var(--color-light);
            min-width: 144px;
            padding: 3px 8px;
            background: var(--color-primary-300);
        }
    }
    .title {
        margin-top: 8px;
        font-style: normal;
        font-weight: 300;
        font-size: 15px;
        line-height: 22px;
        letter-spacing: 0.075px;
        color: var(--color-dark-500);
        overflow: hidden;
        text-overflow: ellipsis;
        display: -webkit-box;
        line-clamp: 2;
        -webkit-line-clamp: 2;
        -webkit-box-orient: vertical;
    }

    .tags {
        display: flex;
        gap: 10px;
        .tag {
            font-style: normal;
            font-weight: 300;
            font-size: 12px;
            line-height: 22px;
            letter-spacing: 0.06px;
            color: var(--color-primary-400);
        }
    }
`;

export {
    WrapColumnItem
}