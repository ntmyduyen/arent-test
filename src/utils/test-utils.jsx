/* eslint-disable import/export */
import setupStore from '@redux/index';
import { cleanup, render } from '@testing-library/react'
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { afterEach } from 'vitest'

afterEach(() => {
	cleanup()
})

function renderWithProviders(
	ui,
	{
		preloadedState = {},
		// Automatically create a store instance if no store was passed in
		store = setupStore(preloadedState),
		...renderOptions
	} = {}
) {
	function Wrapper({ children }) {
		return <Provider store={store}>
			<MemoryRouter>{children}</MemoryRouter>
		</Provider>
	}
	return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
export {
	renderWithProviders
}