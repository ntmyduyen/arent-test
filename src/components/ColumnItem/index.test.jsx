import { renderWithProviders } from "@utils/test-utils"
import ColumnItem from "."

describe('ColumnItem', () => { 
    it('should be match with snapshot', () => {
        const { container } = renderWithProviders(<ColumnItem
            url="test-url"
            dateTime="DD/MM/YYYY HH:mm"
            title="title"
            tags={["tag1", "tag2"]}
        />)
        expect(container).toMatchSnapshot()
    })
})