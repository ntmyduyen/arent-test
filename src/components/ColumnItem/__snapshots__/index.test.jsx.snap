// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ColumnItem > should be match with snapshot 1`] = `
<div>
  <div
    class="sc-beqWaB gICPwG"
  >
    <div
      class="wrap-img"
    >
      <div
        class="date-time"
      >
        DD/MM/YYYY HH:mm
      </div>
    </div>
    <div
      class="title"
      title="title"
    >
      title
    </div>
    <div
      class="tags"
    >
      <div
        class="tag"
      >
        #
        tag1
      </div>
      <div
        class="tag"
      >
        #
        tag2
      </div>
    </div>
  </div>
</div>
`;
